File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.py
def cnt_divisor(n):
    cnt = 1
    for k in range(2,n+1):
        if n%k == 0:
            cnt += 1
    return cnt
    

def solution(left, right):
    answer = 0
    for k in range(left, right+1):
        if (cnt_divisor(k) % 2 == 0):
            answer += k
        else: answer -= k
    
    return answer

File: 프로그래머스/1/77884. 약수의 개수와 덧셈/약수의 개수와 덧셈.java
class Solution {
    public int solution(int left, int right) {
        int result = 0;
        
        for (int i=left; i<right+1; i++) {
            int cnt = 1;
            for (int j=2; j<i+1; j++) {
                if (i%j==0) cnt += 1;
            }
            
            if (cnt%2 == 0) result += i;
            else result -= i;
        }
        
        return result;
    }
}